### PR TITLE
Add webhook and workflow to handle deployment events from GitHub.

### PR DESCRIPTION
**Pull request by jjm**
### Status 

- Complete pack extension

### Description

Add to GitHub pack a webhook, workflow and extra action to process GitHub Deployment events to allow `packs.install` to be called when the enviroment matches that in the configuration. Following the running of `packs.install` GitHub is updated with the status of the deployment..

### Checklist (tick everything that applies)

- [X] Metadata: pack.yaml, icon, structure (required for new packs)
- [X] Version bump (required for changed packs)
- [X] Code linting (required, can be done after the PR checks)
- [X] [Tests](https://docs.stackstorm.com/development/pack_testing.html) (not required but really recommended)

### CODE REVIEW

File: packs/github/README.md [hunk 1]
```diff
@@ -91,3 +91,35 @@ StackStorm webhook handler.
...
+your own config.yaml.
+
+You can then create a deployment via ChatOPS with the following
+command:
+
+```
+@hubot github deployment create me/my_st2_pack description Lets get the feature to production
+```
+
+#### Limiations
```
Comment (Kami) [inline packs/github/README.md]: Minor type -> Limitations :)

File: packs/github/actions/deployment_event.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+---
+name: "deployment_event"
+description: "Process an deployment event from GitHub."
```
Comment (Kami) [inline packs/github/actions/deployment_event.yaml]: Maybe a more description name and description is not such a bad idea.

From just looking at the name / description I don't really have an idea what action is supposed to do. Description from the workflow ("A workflow to process an github deployment event and install a pack if the environment matches.") should be sufficient.

File: packs/github/actions/workflows/deployment_event.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
+---
+version: '2.0'
+
+github.deployment_event:
+  description: "A workflow to procress an github deployment event and install a pack if the enviroment matches."
```
Comment (Kami) [inline packs/github/actions/workflows/deployment_event.yaml]: Minor type - "procress" and "enviroment"

**Comment by Kami** [comment]
There are some small things which can be improved, but overall nice work :+1:

**Comment by jjm** [comment]
@Kami Thanks, that's the above comments been addressed.

**Comment by Kami** [comment]
@jjm Thanks, will have a look once the build finishes :)

**Comment by Kami** [comment]
Looking good, thanks!

**Pull request merged**